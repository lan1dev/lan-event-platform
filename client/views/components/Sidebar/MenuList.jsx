import styled from 'styled-components';

const MenuList = styled.ul`
  list-style-type: none;
  margin-top: 70px;
`;

export default MenuList;
