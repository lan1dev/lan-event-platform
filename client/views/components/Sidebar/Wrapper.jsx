import styled from 'styled-components';

const Wrapper = styled.div`
  min-height: 100%;
  width: 200px;
  background: white;
`;

export default Wrapper;
