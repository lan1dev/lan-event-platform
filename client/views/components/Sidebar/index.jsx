import styled from 'styled-components';

const Sidebar = styled.aside`
  min-height: 100%;
  width: 200px;
  background: white;
`;

export default Sidebar;
