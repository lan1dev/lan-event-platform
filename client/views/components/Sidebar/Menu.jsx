import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import MenuList from './MenuList';

const propTypes = {
  items: PropTypes.arrayOf({ name: PropTypes.string, href: PropTypes.string }).isRequired
};

const Menu = ({ items }) => (
  <Fragment>
    <MenuList>
      {items.map(item => (
        <li key={item.href}>
          <Link to={item.href}>{item.name}</Link>
        </li>
      ))}
    </MenuList>
  </Fragment>
);

Menu.propTypes = propTypes;

export default Menu;
