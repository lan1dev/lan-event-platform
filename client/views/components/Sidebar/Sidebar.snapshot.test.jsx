import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import Sidebar from './index';

describe('Sidebar', () => {
  let props;
  let wrapper;

  const Wrapper = () => {
    if (!wrapper) {
      wrapper = shallow(<Sidebar {...props}>{props.children}</Sidebar>);
    }
    return wrapper;
  };

  // Reset Component and Props before each test.
  beforeEach(() => {
    props = {};
    wrapper = undefined;
  });

  describe('without props', () => {
    it('Should match snapshot', () => {
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });
});
