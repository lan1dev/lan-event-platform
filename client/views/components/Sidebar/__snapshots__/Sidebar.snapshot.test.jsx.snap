// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar without props Should match snapshot 1`] = `
.c0 {
  min-height: 100%;
  width: 200px;
  background: white;
}

<aside
  className="c0"
/>
`;
